packages/in_app_purchase/lib/src/in_app_purchase_tizen_platform.dart

```diff
@@ -140,7 +140,7 @@ class InAppPurchaseTizenPlatform extends InAppPurchasePlatform {
       final String invoiceId =
           billingResultWrapper.payDetails['InvoiceID'] ?? '';
 
-      billingManager
+      await billingManager
```

Comment:
This is logically different. As far as I remember this method (`buyNonConsumable`) should return immediately without waiting for the purchase completion event.

Reply:
I wrapped it with "unwaited()"